docs/generate-time-series/go.rst

```diff
@@ -0,0 +1,374 @@
+
+.. code-block:: js
+
+    func main() {
+        pos, err := getISSPosition()
+        if err != nil {
+            log.Fatal(err)
+        }
+
+        fmt.Println(pos)
+    }
+
+Save all your changes and run the code in the command line:
+
+.. code-block:: console
+
+    $ go run main.go
+
+The result should contain your geo_point string:
+
+.. code-block:: js
+
+    (104.7298, 5.0335)
+
+
+Set up CrateDB
+==============
+
+First, import the `context`_ package from the standard library and the `pgx`_ 
+client:
+
+.. code-block:: js
+
+    import (
+        "context"
+        "encoding/json"
+        "flag"
+        "fmt"
+        "io/ioutil"
+        "log"
+        "net/http"
+
+        "github.com/jackc/pgx/v4"
+    )
+
+Then, in your main function, connect to CrateDB using the `Postgres Wire Protocol`_ port
+(``5432``) and `create a table`_ suitable for writing ISS position coordinates:
+
+.. code-block:: js
+
+    var conn *pgx.Conn
+
+    func main() {
+        var err error
+        conn, err = pgx.Connect(context.Background(), "postgresql://crate@localhost:5432/doc")
+        if err != nil {
+            log.Fatalf("unable to connect to database: %v\n", err)
+        }
+	    defer conn.Close(context.Background())
+
+        conn.Exec(context.Background(), "CREATE TABLE iss (
+                                            timestamp TIMESTAMP GENERATED ALWAYS AS CURRENT_TIMESTAMP, 
+                                            position GEO_POINT)")
+    }
+
+
+Save all your changes and run the new code in the command line:
+
+.. code-block:: console
+
+    $ go run main.go
+
+When you run the script this time, the ``go`` command will look up the module 
+containing the `pgx`_ package and add it to ``go.mod``. In the `CrateDB 
+Admin UI`_, you should see the new table when you navigate to the *Tables* 
+screen using the left-hand navigation menu:
+
+.. image:: ../_assets/img/generate-time-series/table.png
+
+
+Record the ISS position
+=======================
+
+With the table in place, you can start recording the position of the ISS.
+
+Create some logic that calls your ``getISSPosition`` function and `INSERT`_ the
+result into the ``iss`` table:
+
+.. code-block:: js
+
+    ...
+
+    func main() {
+        ...
+
+        pos, err := getISSPosition()
+        if err != nil {
+            log.Fatalf("unable to get ISS position: %v\n", err)
+        } else {
+            _, err := conn.Exec(context.Background(), "INSERT INTO iss (position) VALUES ($1)", pos)
+            if err != nil {
+                log.Fatalf("unable to insert data: %v\n", err)
+            }
+        }
+    }
+
+Save all your changes and run the code in the command line:
+
+.. code-block:: console
+
+    $ go run main.go
+
+Press the up arrow on your keyboard and hit *Enter* to run the same command a
+few more times.
+
+When you're done, you can `SELECT`_ that data back out of CrateDB by executing
```

Comment:
what do you think about including an example of the output here? perhaps you could reformat the table that is displayed using RST tables

[edit: on second thoughts, given that we already include a screenshot of the admin UI elsewhere in these tutorials, perhaps a screenshot would be better than a table)

this also makes me think that the CREATE TABLE command would be easier to run from the Admin UI

Reply:
Yeah I wasn't really sure which to go with (writing logic in the script or using the Admin Console).  In this case, since I am trying to create 1 script, I thought it would make more sense to not include a function that just selects back out all the data and displays it on stdout.  With CREATE TABLE, if I change it to CREATE TABLE [ IF NOT EXISTS ], then maybe it's a good example to show in the code snippet?

Reply:
And given that we already include a screenshot of the admin UI near the end of the tutorial, would it be redundant to include a very similar one again...or would it be consistent?

Reply:
yes, that makes sense. okay! it's fine as is imo :)